src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * duke.Parser that handles all the string manipulation. The users input is taken in as a string and passed to parser
+ * for packaging into a command.
+ */
+public class Parser {
+
+    /**
+     * Function that handles all the parsing of the duke.Parser system
+     *
+     * @param command Refers to the entire line of strings the user entered. The line is then process here
+     *                and packaged into duke.Command. See duke.Command for more.
+     * @return A packaged command for ease of access of infomation
+     */
+    public static Command parse(String command) throws InvalidCommandException {
```

Comment:
Could consider renaming command to userInput, since you will perform checks later to see if it is valid